File: apps/android/app/src/main/java/com/healthmd/domain/model/SleepDayAttribution.kt
package com.healthmd.domain.model

/**
 * Which daily note owns a sleep session (issue #104).
 *
 * Shared cross-platform setting. Apple and Android persist the same [wireValue]
 * strings and default to [NIGHT_BEGINS].
 *
 * @property NIGHT_BEGINS shipped noon-to-noon journaling behavior. Summary
 *   intervals are clipped at the window boundaries; this remains the default
 *   so existing exports never change silently.
 * @property MORNING_ENDS the note for the wake-up date (the calendar date of
 *   the session end) owns the whole session, matching the Health Connect UI.
 */
enum class SleepDayAttribution(val wireValue: String) {
    NIGHT_BEGINS("night_begins"),
    MORNING_ENDS("morning_ends");

    companion object {
        val DEFAULT: SleepDayAttribution = NIGHT_BEGINS

        /** Unknown persisted values fail closed to the shipped default. */
        fun fromWireValue(raw: String?): SleepDayAttribution =
            entries.firstOrNull { it.wireValue == raw } ?: DEFAULT
    }
}

/**
 * Capture-entry choice for sleep attribution.
 *
 * This shape keeps an explicit NIGHT_BEGINS override distinct from "read the
 * stored preference"; the shipped default enum value is never used as a sentinel.
 */
sealed interface SleepDayAttributionOverride {
    data object StoredPreference : SleepDayAttributionOverride
    data class Value(val attribution: SleepDayAttribution) : SleepDayAttributionOverride
}

/** Immutable timezone and sleep-owner policy for one Android capture operation. */
data class AndroidCaptureContext(
    val zoneId: java.time.ZoneId,
    val sleepDayAttribution: SleepDayAttribution,
) {
    val explicitSleepDayAttributionOverride: SleepDayAttributionOverride
        get() = SleepDayAttributionOverride.Value(sleepDayAttribution)
}

File: apps/apple/HealthMd/Shared/Models/SleepDayAttribution.swift
//
//  SleepDayAttribution.swift
//  Health.md
//
//  Issue #104: which daily note owns a sleep session that spans midnight.
//

import Foundation

/// Which daily note owns a sleep session.
///
/// Shared cross-platform setting (issue #104). Both platforms persist the same
/// raw values, default to `nightBegins`, and keep the whole session in a single
/// note; only the owning calendar date differs.
///
/// - `nightBegins`: the note for the calendar date the session starts owns it.
///   This is the shipped noon-to-noon journaling behavior and remains the
///   default so existing exports never change silently.
/// - `morningEnds`: the note for the wake-up date (the calendar date of the
///   session end) owns the whole session, matching the Health Connect UI.
nonisolated enum SleepDayAttribution: String, CaseIterable, Codable, Sendable, Equatable {
    case nightBegins = "night_begins"
    case morningEnds = "morning_ends"

    var localizedDisplayName: String {
        switch self {
        case .nightBegins: return String(localized: "Night begins")
        case .morningEnds: return String(localized: "Morning ends")
        }
    }

    var localizedDescription: String {
        switch self {
        case .nightBegins:
            return String(localized: "Each sleep session belongs to the daily note for the night it starts. This is the default and matches previous exports.")
        case .morningEnds:
            return String(localized: "Each sleep session belongs to the daily note for the morning it ends, matching the Health Connect app. A session from 11:45 PM to 7:30 AM appears in the wake-up day's note.")
        }
    }
}
